HPCC-16287 Add roxie-centric _TransactionId to request top level

rpastrana opened the pull request:
- Adds _TransactionId to Roxie outbound requests
- Adds _TransactionId to Roxie test tool

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@afishbeck please review.

afishbeck commented on esp/services/esdl_svc_engine/esdl_binding.cpp:
trivial, but doesn't seem like a blank line is appropriate here.

afishbeck reviewed (approved):
@rpastrana Other than extra blank line, looks good.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 334ec5e7235f179f8ab76679e6904bc79e471793
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 96.072 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |9 |9 |0 |
|setup (thor) |9 |9 |0 |
|setup (roxie) |9 |9 |0 |
|test (hthor) |720 |720 |0 |
|test (thor) |615 |615 |0 |
|test (roxie) |745 |745 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

rpastrana committed:
HPCC-16287 Add roxie-centric _TransactionId to request top level

- Adds _TransactionId to Roxie outbound requests
- Adds _TransactionId to Roxie test tool

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@afishbeck superfluous new line removed
@richardkchapman please merge

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: ab54297bcd14e008c30f1c514a9b7bc1805f743a
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.866 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |9 |9 |0 |
|setup (thor) |9 |9 |0 |
|setup (roxie) |9 |9 |0 |
|test (hthor) |720 |719 |1 |
|test (thor) |615 |615 |0 |
|test (roxie) |745 |744 |1 |


HPCC Stop: OK
HPCC Uninstall: OK
